### ISSUE TITLE
How do I make the leftover liquid back to the tank?

### ISSUE THREAD

**Issue by Lion4K** [rwtema/extrautilities#1014]
Like, i'm using TConstruct, the SearedSmeltery can have multiple fluids stored at once.

If I put a Transfer Node (fluid) attached to the Drain, it will transfer the content to the drum.

It will fill the drum up to the smeltery content and after that liquid is all gone, it will start gathering inside of it the next fluid type, 1000mb of it.

Now lets say I want that liquid in the smeltery and not in the node, how to I make the node semi automatic or stop it from gathering those said 1000mb?

Redstone Signal prevents it from sucking more liquid, but as well from letting it go, so I can't really make a loop back to the furnace.

**Comment by xChitose** [comment]
Transfer nodes will fill up if the inventory connected to it allows it (auto-outputs). 

You could try two things, the first one would be a retrieval node on another part of the shelter (two drains, if you will, one leading back in, and the other leading out), and have a constant red stone signal on the node which you can trigger whenever you want to empty the original node (this would put the 1,000mb of the fluid that gets "stuck" in the node back into the smeltery, unless you'd like to put it anywhere else) 

But another method would be to use a filter on the node, which is a tad impractical since you'd have to setup a filter every time you did a smeltery operation, unless you had a filter system for every metal you own that the shelter accepts. 

These are just some suggestions which I cannot test right now, but let me know how it goes!

**Comment by Lion4K** [comment]
The filter thing, i've eventually figured, but as you said it's not practical. I'm gonna test the retrieval one.

**Comment by xChitose** [comment]
Let me know how it works!

**Comment by Lion4K** [comment]
It didn't worked.

The retriev one didn't worked either.

The liquid keeps stuck inside of it.

Or i'm not wiring it properly

**Comment by xChitose** [comment]
Maybe world interaction upgrades?

**Comment by xChitose** [comment]
When I get to my computer, I'll give it a try.

**Comment by xChitose** [comment]
Do you have any other mods that add fluid pipes? (Ender IO, Mekanism, etc.)

**Comment by Lion4K** [comment]
No, only Extrautils, I never found any simple piping system, I'm playing TinkersConstruct and Buildcraft would add too much stuff to get into piping idk.

**Comment by Lion4K** [comment]
actually idrk cuz last time I used builcraft was when ModloaderMP was in need

**Comment by xChitose** [comment]
Buildcraft 7 is real nice in terms of combining all of the add-ons into the main branch. 

But, if you ever consider getting any other tech mods that add piping then let me know, it could call for a nice and easy system. 

for example, Ender IO has cables that can fit in one block space, so to say you could have a liquid and power going into the back of one block. 

It also has a pipe that can transfer multiple liquids through the same pipe (making that system for multiple drum setups a little less impractical) and of course filtering.

**Comment by xChitose** [comment]
Also, I did a test with using nodes and the only thing I could think of is having a buffer drum (or tank) and having an input/output on that with a relative signal, so instead of having the alloy get held in the node, it would instead be held in the drum and then this could lead back into the smeltery, but this could get sticky if you're dealing with multiple molten metals.

**Comment by Lion4K** [comment]
Oh aparently BuildCraft is now modular ?! means I can get only the piping.
Thou seems not availble to 1.10.2

**Comment by Lion4K** [comment]
Yeah. I think the only option would be to make the node w/o bufffer and make buffer upgrades do this job, and therefore only when really needed.

**Comment by xChitose** [comment]
Yeah, buildcraft isn't here on 1.10.2 yet. 

Also, I'm going to play around with it a bit and see if I can figure anything out.

**Comment by Lion4K** [comment]
EnderIO on the other hand, seems like might work. gonna test it.

**Comment by xChitose** [comment]
Yeah, pretty much with EIO, since you can wrench the base of the pipe, any liquid that happens to get "stuck" will just be thrown back into the source (the smeltery, for example).

**Comment by Lion4K** [comment]
ooh goddammit, EnderIO says "optional" libs JEI, then EnderIO require JEI xxxxxx Install or remove EnderIO xD

**Comment by xChitose** [comment]
Is JEI outdated?

**Comment by Lion4K** [comment]
It was, 26xsomething asking for 27xsomething. It now have 3xxsomething xD

**Comment by Lion4K** [comment]
hmm, Thou Ender IS seems too complex for this.

Like,

I consider mods being in 2 categories:

The ones that add a lot of stuff and usually require you to play it, like BC, IC², EIO, TC,

And the ones that add some stuff that can be used to enhance the above mods, like addons, like Extra Utils, JEI, Chicken Bones Wireless Redstone, Redstone Paste, etc.

And at the moment i'm playing TC, and ExtraUt is doing the job, I think i'm gonna stick to it till the end.

**Comment by xChitose** [comment]
Ahh, I got you. 

Well, once you get drums of every alloy you have (with nodes and redstone signals on them all) you'll be in good shape, since all you'll have to do to mix ingots will be to flip a lever. 

And if you want, say, precise input from the node / drum, you could use repeaters that hold it, say 20 ticks and input 2,000mb (just an estimate).

The only tedious part would be getting that many drums.

**Comment by Lion4K** [comment]
In fact I got drums already, but instead,i'm just using them to drain the smeltery dry and then re add the alloys when needed

**Comment by Lion4K** [comment]
I Just came up with some ingenious idea:

Liquid Filter > ✓ Blacklist X Ignore NBT

Few mb's of something you will never need to drain out, Inside the Smelter (I used Liquid Dirt)

A bucket filled with the same liquid and to be inserted in the Filter.

Node, redstone blocked, with filter inserted, just make sure that the chosen fluid is the next in the line of your Smelter and voila.
